src/Components/test/E2ETestMigration/Infrastructure/ServerTestBase.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.BrowserTesting;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.Testing;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETest.Infrastructure
+{
+    public abstract class ServerTestBase<TServerFixture>
+        : ComponentBrowserTestBase,
+        IClassFixture<TServerFixture>
+        where TServerFixture: ServerFixture
+    {
+        public string ServerPathBase => "/subdir";
+
+        protected readonly TServerFixture _serverFixture;
+
+        public ServerTestBase(
+            TServerFixture serverFixture,
+            ITestOutputHelper output)
+            : base(output)
+        {
+            _serverFixture = serverFixture;
+            MountUri = _serverFixture.RootUri + "subdir";
+        }
+
+        public void Navigate(string relativeUrl, bool noReload = false)
+        {
+            //Browser.Navigate(_serverFixture.RootUri, relativeUrl, noReload);
```

Comment:
Do we need to replace this with a Playwright API?

Reply:
I'm not sure exactly what the E2e tests were using this call for, as it sounds like it was a optimization for performance:

https://github.com/dotnet/aspnetcore/blob/0bb4cc8595e17c93ed3db13d7af61d716472338d/src/Components/test/E2ETest/Tests/BindTest.cs#L32

```
            // On WebAssembly, page reloads are expensive so skip if possible
```

I left this here for now as a reminder to encourage discussion like this, as I don't have enough context to know if this is something we should/need to keep doing

Reply:
I sort of replaced the Navigate call with the MountUri property which get's used as part of InitializeCore, so basically every test does a page load, there's no playwright page instance reuse across tests here since I wanted to start from the most separate state, as opposed to trying to reuse anything to see if this helps from a stability standpoint

https://github.com/dotnet/aspnetcore/blob/328910d83b56127165e94cc6e3db67ea799d154c/src/Components/test/E2ETestMigration/Infrastructure/ComponentTestBase.cs#L64

Reply:
In playwright world is fine if we create a new context every time, they are meant to be isolated from each other.